src/Microsoft.IdentityModel.Tokens/BaseConfigurationManager.cs

```diff
@@ -76,6 +111,24 @@ public BaseConfiguration LastKnownGoodConfiguration
             {
                 _lastKnownGoodConfiguration = value ?? throw LogHelper.LogArgumentNullException(nameof(value));
                 _lastKnownGoodConfigFirstUse = DateTime.UtcNow;
+
+                if (_lkgConfigurationCache == null)
```

Comment:
I don't see a reason to delay the creation of the cache, might as well new it up in the ctor or assign the private variable.